src/OctoshiftCLI.Tests/Octoshift/Services/OctoLoggerTests.cs

```diff
@@ -63,29 +63,72 @@ public void Secrets_Should_Be_Masked_From_Logs_And_Console()
     [Fact]
     public void Ghes_Archive_Url_Tokens_Should_Be_Replaced_In_Logs_And_Console()
     {
-
         var ghesArchiveUrl = "https://files.github.acmeinc.com/foo?token=foobar";
-
-        _octoLogger.Verbose = false;
-        _octoLogger.LogInformation($"Archive URL: {ghesArchiveUrl}");
-        _octoLogger.LogVerbose($"Archive URL: {ghesArchiveUrl}");
-        _octoLogger.LogWarning($"Archive URL: {ghesArchiveUrl}");
-        _octoLogger.LogSuccess($"Archive URL: {ghesArchiveUrl}");
-        _octoLogger.LogError($"Archive URL: {ghesArchiveUrl}");
-        _octoLogger.LogError(new OctoshiftCliException("Archive URL: {ghesArchiveUrl}"));
-        _octoLogger.LogError(new InvalidOperationException("Archive URL: {ghesArchiveUrl}"));
-
-        _octoLogger.Verbose = true;
-        _octoLogger.LogVerbose("Archive URL: {ghesArchiveUrl}");
-
-        _consoleOutput.Should().NotContain(ghesArchiveUrl);
-        _logOutput.Should().NotContain(ghesArchiveUrl);
-        _verboseLogOutput.Should().NotContain(ghesArchiveUrl);
-        _consoleError.Should().NotContain(ghesArchiveUrl);
+        var variants = new[]
+        {
+            ghesArchiveUrl,
+            ghesArchiveUrl.ToUpper(),
+            ghesArchiveUrl.ToLower()
+        };
+
+        foreach (var variant in variants)
+        {
+            _octoLogger.Verbose = false;
+            _octoLogger.LogInformation($"Archive URL: {variant}");
+            _octoLogger.LogVerbose($"Archive URL: {variant}");
+            _octoLogger.LogWarning($"Archive URL: {variant}");
+            _octoLogger.LogSuccess($"Archive URL: {variant}");
+            _octoLogger.LogError($"Archive URL: {variant}");
+            _octoLogger.LogError(new OctoshiftCliException($"Archive URL: {variant}"));
+            _octoLogger.LogError(new InvalidOperationException($"Archive URL: {variant}"));
+
+            _octoLogger.Verbose = true;
+            _octoLogger.LogVerbose($"Archive URL: {variant}");
+
+            _consoleOutput.Should().NotContain(variant);
+            _logOutput.Should().NotContain(variant);
+            _verboseLogOutput.Should().NotContain(variant);
+            _consoleError.Should().NotContain(variant);
+        }
```

Comment:
Generally speaking a good unit test needs to test one thing rather than many different things, while sometimes there could be more than one assertion (e.g. asserting some fields in a payload), but still the preferred way should be acting on one thing and asserting the result. The reason is in case of a failure it would be easy to figure out what failed and why.

Now here this test has already been doing too much and was acting on different methods (`LogInformation`, `LogSuccess`, ...), and now it uses a loop to test against more inputs. This will make the test very hard to understand and debug. 

Fortunately `xUnit` has a way to deal with different input values w/o having to write more than one test or using loops and such. To do so you need to use `Theory` instead of a `Fact` and pass some input values with `InlineData`, read further [here](https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/).

With a `Theory` your test will look like this:

```c#
[Theory]
[InlineData("https://files.github.acmeinc.com/foo?token=foobar")]
[InlineData("HTTPS://FILES.GITHUB.ACMEINC.COM/FOO?TOKEN=FOOBAR")]
public void Ghes_Archive_Url_Tokens_Should_Be_Replaced_In_Logs_And_Console(string url)
{
    ...
}
```

One last thing is, while there are many school of thoughts, I believe a unit test shouldn't try to be smart, yes sometimes for ease of maintenance we create base classes and private methods mostly for setup and tear down tasks but other than that anything else will make a unit test less maintainable IMO. For example using loops in a test, or using `input.ToUppoer` instead of simply using the upper cased string literal, etc.

Reply:
Done! 🏁